src/db.c

```diff
@@ -123,8 +123,9 @@ robj *lookupKey(serverDb *db, robj *key, int flags) {
         /* Update the access time for the ageing algorithm.
          * Don't do it if we have a saving child, as this will trigger
          * a copy on write madness. */
+        client *executing_client = (server.executing_client) ? server.executing_client : server.current_client;
         if (server.current_client && server.current_client->flag.no_touch &&
-            server.current_client->cmd->proc != touchCommand)
+            executing_client->cmd->proc != touchCommand)
```

Comment:
executing_client is always true in this case i think, i guess we can just use it.
```suggestion
            server.executing_client->cmd->proc != touchCommand)
```

Reply:
I wasn't sure about that. Changed.